### PR TITLE
Update Dockerfile to fix `ip` binary issue

### LINKED ISSUE #70: Default gh_actions_host domain is no longer valid?
> I've noticed that the docker image tests have been failing for a few days (see e.g. https://github.com/Materials-Consortia/optimade-python-tools/pull/905/checks?check_run_id=3381031009). It looks like Actions may have removed the `gh_actions_host` access to "localhost"? Any ideas on this @CasperWA?

*Originally posted by @ml-evs in https://github.com/Materials-Consortia/optimade-python-tools/issues/906#issue-975484312*

### PR DESCRIPTION
**Pull request by CasperWA**
Fixes #70.

- [X] Add pipefail command to Dockerfile.
- [x] Set specific docker image and install minimum dependencies.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,7 @@
 FROM python:3.9
```
Comment (ml-evs) [inline Dockerfile:1]:
```suggestion
FROM python:3.9-alpine
```

File: Dockerfile [hunk 2]
```diff
@@ -1,6 +1,7 @@
...
 
-RUN echo $(ip route | awk '{print $3}') > /docker_host_ip
+RUN set -o pipefail \
```
Comment (ml-evs) [inline Dockerfile:3]: If you want to keep the pipefail, I found I had do to this:

```suggestion
RUN ["/bin/bash", "-c", "set -o pipefail && ip route | awk '{print $3}' > docker_host_ip"]
```

File: Dockerfile (hunk 1 above)
Comment (ml-evs) [reply Dockerfile:1]: or

```suggestion
FROM python:3.9-slim-buster
```

File: Dockerfile (hunk 2 above)
Comment (CasperWA) [reply Dockerfile]: Right - if the current changes don't work I'll do this instead.

**Review by ml-evs — APPROVED** [review]
Thanks @CasperWA!

**Pull request merged**